include/dali/core/span.h

```diff
@@ -58,10 +59,30 @@ class span {
   >>
   DALI_HOST_DEV constexpr span(const span<U, Extent>& s) noexcept : data_(s.data()) {}
 
+
   DALI_HOST_DEV constexpr span(pointer firstElem, pointer lastElem) : data_(firstElem) {
     /* assert(lastElem - firstElem == Extent); */
   }
 
+  /**
+   * @remarks Container is not a specialization of span,
+   *          specialization of std::array or an array type
+   */
+  template <typename Container,
+          typename std::enable_if<is_container<Container>::value, int>::type = 0>
+  DALI_HOST_DEV constexpr span(Container &container)
+          : span(container.data(), container.size()) {}
+
+  /**
+   * @remarks Container is not a specialization of span,
+   *          specialization of std::array or an array type
+   */
+  template <typename Container,
+          typename std::enable_if<is_container<Container>::value, int>::type = 0>
+  DALI_HOST_DEV constexpr span(const Container &container)
```

Comment:
That's the part I dislike the most - const-ref constructor. Use universal reference (vide `make_span`) and explicitly reject r-values.